src/Features/Core/Portable/ConvertAnonymousTypeToClass/AbstractConvertAnonymousTypeToClassCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,412 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.GenerateEqualsAndGetHashCodeFromMembers;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertAnonymousTypeToClass
+{
+    internal abstract class AbstractConvertAnonymousTypeToClassCodeRefactoringProvider<
+        TExpressionSyntax,
+        TIdentifierNameSyntax,
+        TObjectCreationExpressionSyntax,
+        TAnonymousObjectCreationExpressionSyntax,
+        TNamespaceDeclarationSyntax> :
```

Comment:
> : [](start = 36, length = 2)

Nit: in previous PRs you had the colon on next line, I think.
Either way, I found this a bit hard to parse visually.

Reply:
Will tweak.  Thanks for the feedback on this!  I would def like a formatting option to place hte ```:``` for inheritance or ```:this/base``` on the next line when multi-line.

Reply:
Done.